src/d/actor/d_a_kita.cpp

```diff
@@ -4,52 +4,448 @@
 //
 
 #include "d/actor/d_a_kita.h"
+#include "d/actor/d_a_player.h"
 #include "d/d_bg_w.h"
+#include "d/d_com_inf_game.h"
 #include "d/d_procname.h"
+#include "d/d_s_play.h"
+#include "f_op/f_op_actor_mng.h"
+#include "d/d_bg_s_movebg_actor.h"
+
+// Inline helper because i cannot get things to match otherwise
+inline void cap_min_val(float& a, float b){
+    if(a < b) a = b;
+}
+
 
 /* 00000078-0000032C       .text ride_call_back__FP4dBgWP10fopAc_ac_cP10fopAc_ac_c */
-void ride_call_back(dBgW*, fopAc_ac_c*, fopAc_ac_c*) {
-    /* Nonmatching */
+void ride_call_back(dBgW* bgw, fopAc_ac_c* i_ac, fopAc_ac_c* i_pt) {
+    kita_class* pActor = static_cast<kita_class*>(i_ac);
+
+    cXyz delta_pos, local_44, local_50;
+    if ((pActor->field_29A == 0) || (pActor->field_360 != 0)) {
+        mDoMtx_YrotS(*calc_mtx, -pActor->current.angle.y);
+        delta_pos = i_pt->current.pos - pActor->current.pos;
+        MtxPosition(&delta_pos,&local_44);
+        delta_pos = i_pt->old.pos - pActor->current.pos;
+        MtxPosition(&delta_pos, &local_50);
+        if (fopAcM_GetName(i_pt) == 169) {
+            pActor->mExecuteCount = 10;
+        }
+        short zAngle_target = -local_44.x * ((REG0_F(0) + 10.0f) / pActor->scale.x);
+        short xAngle_target = local_44.z * ((REG0_F(0) + 10.0f) / pActor->scale.z);
+        cLib_addCalcAngleS2(&pActor->current.angle.x,xAngle_target,10,0x800);
+        cLib_addCalcAngleS2(&pActor->current.angle.z,zAngle_target,10,0x800);
+        cap_min_val(pActor->field_2BC.x,  (REG0_F(4) + 50.0f) * std::abs(local_44.z - local_50.z));
+        cap_min_val(pActor->field_2BC.z, (REG0_F(4) + 50.0f) * std::abs(local_44.x - local_50.x));
```

Comment:
What was the specific issue you were having that you had to add the fake inline? Could you post a scratch?